packages/git/src/common/git.ts

```diff
@@ -691,12 +703,20 @@ export interface Git extends Disposable {
     show(repository: Repository, uri: string, options?: Git.Options.Show): Promise<string>;
 
     /**
-     * It resolves to an array of configured remotes for the given repository.
+     * It resolves to an array of configured remotes names for the given repository.
      *
-     * @param repository the repository to get the remotes.
+     * @param repository the repository to get the remote names.
      */
     remote(repository: Repository): Promise<string[]>;
 
+    /**
+     * It resolves to an array of configured remote objects for the given Git action.
+     *
+     * @param repository the repository to get the remote objects.
+     * @param options `git remote` command refinements.
+     */
+    remote(repository: Repository, options: Git.Options.Remote): Promise<RemoteModel[]>;
```

Comment:
It should be:
```ts
remote(repository: Repository, options: { verbose: true }): Promise<RemoteModel[]>;
```

With your approach I can do the following on the call-site:
```ts
await this.git.remote({ localUri: 'file:///blabla' }, {}); // {} is valid becasue `verbose` is optional. We must require an explicit `verbose = true`. Otherwise, it won't work.
```

But please correct me if I have overlooked something.

Reply:
yes I agree, I've fixed it